internal/controller/pipelinerollout/pipelinerollout_progressive.go

```diff
@@ -472,48 +550,61 @@ func createScaledDownUpgradingPipelineDef(
 		return err
 	}
 
-	// map each vertex name to new min/max, which is based on the number of Pods that were removed from the corresponding
```

Comment:
none of this code below has changed - it's just indented because of a new if/else in front of it